src/CheckStrings.jl

```diff
@@ -0,0 +1,200 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+## Functions to check validity of UTF-8, UTF-16, and UTF-32 encoded strings,
+#  and also to return information necessary to convert to other encodings
+
+module CheckStrings
+
+using Encodings
+
+export check_string
+export is_surrogate_lead, is_surrogate_trail, is_surrogate_codeunit, is_valid_continuation
+
+using Base.UTF_ERR_SHORT, Base.UTF_ERR_CONT,Base.UTF_ERR_LONG,
+      Base.UTF_ERR_NOT_LEAD, Base.UTF_ERR_NOT_TRAIL,
+      Base.UTF_ERR_SURROGATE, Base.UTF_ERR_MISSING_SURROGATE, Base.UTF_ERR_INVALID
+
+is_surrogate_lead(c::Unsigned) = ((c & ~0x003ff) == 0xd800)
+is_surrogate_trail(c::Unsigned) = ((c & ~0x003ff) == 0xdc00)
+is_surrogate_codeunit(c::Unsigned) = ((c & ~0x007ff) == 0xd800)
+is_valid_continuation(c) = ((c & 0xc0) == 0x80)
```

Comment:
The world is your oyster! (particularly for an overhaul, even if it's still a slightly obscure package)